hypertrace-graphql-labels-schema-impl/src/main/java/org/hypertrace/graphql/label/dao/LabelApplicationRuleRequestConverter.java

```diff
@@ -133,6 +133,9 @@ StringCondition convertStringCondition(
     StringCondition.Builder stringConditionBuilder =
         StringCondition.newBuilder()
             .setOperator(convertStringConditionOperator(stringCondition.operator()));
+    if (stringCondition.stringConditionValueType() == null) {
```

Comment:
When `stringCondition.stringConditionValueType()` was null the below switch case was throwing NPE. Earlier I thought default will handle null case as well.

Reply:
Rather than duplicating the return here, use something like `Type type = Optional.ofNullable(condition.stringConditionValueType()).orElse(VALUE);`, and then send it into the switch. Also make sure the valueType is annotated `@Nullable`, we do that so the IDE should have flagged the existing code.

Reply:
It is already annotated with `@Nullable` here: https://github.com/hypertrace/hypertrace-graphql/blob/main/hypertrace-graphql-labels-schema-api/src/main/java/org/hypertrace/graphql/label/schema/rule/StringCondition.java#L50

Reply:
Gotcha - make sure your IDE is set up to highlight the issue. We really need to update our CI-run static analysis to check things like this too.